packages/go_router/lib/src/parser.dart

```diff
@@ -72,7 +72,7 @@ class GoRouteInformationParser extends RouteInformationParser<RouteMatchList> {
       return debugParserFuture = _redirect(context, matchList)
           .then<RouteMatchList>((RouteMatchList value) {
         if (value.isError && onParserException != null) {
-          return onParserException!(context, value);
+          return onParserException!(context, value); // ignore: use_build_context_synchronously
```

Comment:
@chunhtai Here and below I am actually not sure what should happen if the context has become invalid. Do you have a suggestion?

Reply:
if the context is invalid, it should just return a never complete future.

Something like `return Completer<RouteMatchList>().future`